### PR TITLE
Disable or fix some spammy new clang-tidy 13 warnings

### PR DESCRIPTION
**Pull request by def-**
```
src/engine/client/backend/opengles/opengles_sl_program.cpp:10:11: warning: suspicious #include of file with '.cpp' extension [bugprone-suspicious-include]
          ^

src/engine/graphics.h:330:67: warning: 2 adjacent parameters of 'CFreeformItem' of similar type ('float') are easily swapped by mistake [bugprone-easily-swappable-parameters]
                CFreeformItem(float x0, float y0, float x1, float y1, float x2, float y2, float x3, float y3) :
                                                                                ^~~~~~~~~~~~~~~~~~

src/game/editor/layer_tiles.cpp:1802:14: warning: result of multiplication in type 'int' is used as a pointer offset after an implicit widening conversion to type 'ptrdiff_t' [bugprone-implicit-widening-of-multiplication-result]
                        mem_zero(&m_pSwitchTile[y * m_Width], o * sizeof(CSwitchTile));
                                  ^

src/game/server/entities/plasma.cpp:100:152: warning: operator has equivalent nested operands [misc-redundant-expression]
        if(SnapPlayer && SnapPlayer->GetTeam() != TEAM_SPECTATORS && !SnapPlayer->IsPaused() && SnapChar && SnapChar && SnapChar->Team() != m_ResponsibleTeam && SnapPlayer->m_ShowOthers != 1)
                                                                                                ~~~~~~~~    ~~~~~~~~                                          ^

src/game/editor/editor.cpp:4295:38: warning: The right operand of '>' is a garbage value [clang-analyzer-core.UndefinedBinaryOperatorResult]
                        if(m_FilePreviewImageInfo.m_Width > Preview.w)
                                                          ^ ~~~~~~~~~
```

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/skins.cpp [hunk 1]
```diff
@@ -24,14 +24,7 @@ static const char *VANILLA_SKINS[] = {"bluekitty", "bluestripe", "brownbear",
...
 {
-	for(auto &pVanillaSkin : VANILLA_SKINS)
-	{
-		if(str_comp(pName, pVanillaSkin) == 0)
-		{
-			return true;
-		}
-	}
-	return false;
+	return std::all_of(std::begin(VANILLA_SKINS), std::end(VANILLA_SKINS), [pName](const char *pVanillaSkin) { return str_comp(pName, pVanillaSkin) == 0; });
```
Comment (Robyt3) [inline src/game/client/components/skins.cpp:27]: This changes the semantics. Should probably be `any_of` instead of `all_of`.

Comment (def-) [reply src/game/client/components/skins.cpp:27]: Done

**Comment by def-** [comment]
I picked the checks that sounded good to me, if someone has different opinions, say so. For reference: https://clang.llvm.org/extra/clang-tidy/checks/list.html

**Comment by heinrich5991** [comment]
Looks mostly good to me, some of the `else` stuff getting removed decreases symmetry, but it's not too bad.

What do you think about the graphics code changes, @Jupeyy?

**Comment by Jupeyy** [comment]
> Looks mostly good to me, some of the `else` stuff getting removed decreases symmetry, but it's not too bad.
> What do you think about the graphics code changes, @Jupeyy?

yeah same argument, but code look good to me

File: src/engine/client/backend_sdl.h [hunk 2]
```diff
@@ -143,9 +143,9 @@ class CCommandProcessorFragment_SDL
 
 private:
 	void Cmd_Init(const SCommand_Init *pCommand);
-	void Cmd_Shutdown(const SCommand_Shutdown *pCommand);
+	static void Cmd_Shutdown(const SCommand_Shutdown *pCommand);
```
Comment (Jupeyy) [inline src/engine/client/backend_sdl.h:146]: This for example is a bit "overkill", sure they could be static rn
But when they are implemented some day it has to be removed again.

But ok xd idc that much tbh

**Comment by Jupeyy** [comment]
Thinking about it a bit, I wonder if forcing this in CI doesn't make new contributions alot harder for almost no gain, especially the readability stuff. Do others think it's generally worth it?

**Comment by def-** [comment]
I can remove the else-removal and static checks.

The contributors seem to struggle with clang-format already. I'm ok with just letting stuff in and I can run a clang-tidy/clang-format fixit over it. I don't know if that's keeping anyone from contributing to ddnet.

**Comment by heinrich5991** [comment]
Yeah, I think I'd be in favor of removing the static/else checks.

File: src/game/client/components/emoticon.cpp [hunk 3]
```diff
@@ -181,16 +181,16 @@ void CEmoticon::OnRender()
 	Graphics()->WrapNormal();
 }
 
-void CEmoticon::Emote(int Emoticon)
+void CEmoticon::Emote(int Emote)
```
Comment (Jupeyy) [inline src/game/client/components/emoticon.cpp:184]: I wonder how this goes through clang tidy and the compiler

Comment (Jupeyy) [reply src/game/client/components/emoticon.cpp:184]: maybe -Wshadow, the best compiler flag to not use `int i` twice xD

File: src/engine/server/databases/sqlite.cpp [hunk 4]
```diff
@@ -224,12 +224,12 @@ void CSqliteConnection::BindFloat(int Idx, float Value)
 
 // TODO(2020-09-07): remove extern declaration, when all supported systems ship SQLite3 version 3.14 or above
```
Comment (heinrich5991) [inline src/engine/server/databases/sqlite.cpp:225]: Either the TODO needs to be removed (and the empty `#if defined`) or the symbol should be re-added.

File: src/game/server/ddracechat.cpp [hunk 5]
```diff
@@ -1294,9 +1294,7 @@ void CGameContext::ConSpecTeam(IConsole::IResult *pResult, void *pUserData)
 
 bool CheckClientID(int ClientID)
 {
-	if(ClientID < 0 || ClientID >= MAX_CLIENTS)
-		return false;
-	return true;
+	return !(ClientID < 0 || ClientID >= MAX_CLIENTS);
```
Comment (heinrich5991) [inline src/game/server/ddracechat.cpp:1297]:
```suggestion
	return 0 <= ClientID && ClientID < MAX_CLIENTS;
```

**Comment by heinrich5991** [comment]
LGTM except the two nits.

File: src/engine/server/databases/sqlite.cpp (hunk 4 above)
Comment (def-) [reply src/engine/server/databases/sqlite.cpp:225]: Done.

File: src/game/server/ddracechat.cpp (hunk 5 above)
Comment (def-) [reply src/game/server/ddracechat.cpp:1297]: Done.

**Comment by heinrich5991** [comment]
bors r+

Thanks. :)

**Pull request merged**